File: An II/Sem I/MAP/Seminar/Sem1/src/Runners/StrategyTaskRunner.java
package Runners;

import Container.Container;
import Enums.Strategy;
import Factory.TaskContainerFactory;
import Model.Task;

public class StrategyTaskRunner implements TaskRunner{
    private Container container;

    public StrategyTaskRunner(Strategy strategy) {
        this.container = TaskContainerFactory.getInstance().createContainer(strategy);
    }

    @Override
    public void executeOneTask() {
        Task task = container.remove();
        task.execute();
    }

    @Override
    public void executeAll() {
        while(hasTask())
            executeOneTask();
    }

    @Override
    public void addTask(Task t) {
        container.add(t);
    }

    @Override
    public boolean hasTask() {
        return !container.isEmpty();
    }
}

File: An II/Sem I/MAP/Lab/Lab11/Lab11/Runners/StrategyTaskRunner.cs
using Lab11.Enums;
using Lab11.Factory;
using Lab11.Container;

namespace Lab11.Runners;

public class StrategyTaskRunner : TaskRunner
{
    
    private Container.Container container;

    public StrategyTaskRunner(Strategy strategy)
    {
        container = TaskContainerFactory.getInstance.createContainer(strategy);
    }
    
    public void executeOneTask()
    {
        if (!hasTask()) return;
        Task? task = container.remove();
        task.execute();
    }

    public void executeAll()
    {
        while (hasTask())
        {
            executeOneTask();
        }
    }
    
    public bool hasTask()
    {
        return !container.isEmpty();
    }
    
    public void addTask(Task? t)
    {
        container.add(t);
    }
}